src/peer_select.cc

```diff
@@ -132,6 +252,23 @@ PeerSelector::~PeerSelector()
     delete lastError;
 }
 
+void
+PeerSelector::startPingWaiting()
+{
+    assert(entry);
+    assert(entry->ping_status != PING_WAITING);
+    ThePeerSelectorPingMonitor.monitor(this);
+    entry->ping_status = PING_WAITING;
+}
+
+void
+PeerSelector::cancelPingTimeoutMonitoring()
+{
+    assert(entry);
+    assert(entry->ping_status == PING_WAITING);
+    ThePeerSelectorPingMonitor.forget(this);
```

Comment:
```suggestion
    ThePeerSelectorPingMonitor.forget(this);
    entry->ping_status = PING_DONE;
```
I would add this to make it symmetrical with startPingWaiting() and remove that line from PeerSelector::selectMore().

Reply:
This asymmetry is intentional. Please see cb5bc1c.